diff -Nur -x '*.orig' -x '*~' libgems-ruby/lib/rubygems/uninstaller.rb libgems-ruby.new/lib/rubygems/uninstaller.rb
--- libgems-ruby/lib/rubygems/uninstaller.rb	2008-07-16 08:40:59.000000000 +0100
+++ libgems-ruby.new/lib/rubygems/uninstaller.rb	2008-07-16 08:42:37.000000000 +0100
@@ -90,8 +90,9 @@
       hook.call self
     end
 
-    remove spec, specs
     specs.each do |s| remove_executables s end
+    remove spec, specs
+
 
     Gem.post_uninstall_hooks.each do |hook|
       hook.call self
