diff -Nur -x '*.orig' -x '*~' libgems-ruby/lib/rubygems/commands/update_command.rb libgems-ruby.new/lib/rubygems/commands/update_command.rb
--- libgems-ruby/lib/rubygems/commands/update_command.rb	2008-08-13 18:49:11.000000000 +0100
+++ libgems-ruby.new/lib/rubygems/commands/update_command.rb	2008-08-13 18:58:03.000000000 +0100
@@ -44,6 +44,16 @@
     "#{program_name} GEMNAME [GEMNAME ...]"
   end
 
+  def update_rubygems_command
+    "apt-get install rubygems" <<
+    case RUBY_VERSION
+    when /1\.9/
+      "1.9"
+    when /1\.8/
+      "1.8"
+    end
+  end
+
   def execute
     hig = {}
 
@@ -54,6 +64,8 @@
         fail "No gem names are allowed with the --system option"
       end
 
+      exec update_rubygems_command
+
       spec = Gem::Specification.new
       spec.name = 'rubygems-update'
       spec.version = Gem::Version.new Gem::RubyGemsVersion
